conans/model/conan_file.py

```diff
@@ -50,6 +50,22 @@ def create_requirements(conanfile):
         raise ConanException("Error while initializing requirements. %s" % str(e))
 
 
+def create_build_requirements(conanfile):
```

Comment:
I think you don't need to change conan_file.py at all. 
This PR should affect only to the text version of the conanfile. Conanfile.py already manages build_requires, but in a different place.

Reply:
I will remove the changes from these file